contrib/parseq-restli-client/src/main/java/com/linkedin/restli/client/ParSeqRestClient.java

```diff
@@ -0,0 +1,214 @@
+/*
+   Copyright (c) 2012 LinkedIn Corp.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package com.linkedin.restli.client;
+
+import com.linkedin.common.callback.Callback;
+import com.linkedin.parseq.Task;
+import com.linkedin.parseq.batching.Batch;
+import com.linkedin.parseq.batching.BatchImpl.BatchEntry;
+import com.linkedin.parseq.batching.BatchingStrategy;
+import com.linkedin.parseq.promise.Promise;
+import com.linkedin.parseq.promise.Promises;
+import com.linkedin.parseq.promise.SettablePromise;
+import com.linkedin.r2.message.RequestContext;
+import com.linkedin.restli.common.OperationNameGenerator;
+
+
+public class ParSeqRestClient
+    extends BatchingStrategy<RequestGroup, ParSeqRestClient.RestRequestBatchKey, Response<Object>> {
```

Comment:
it's a bit weird for a Client to extend Strategy... maybe we can make the strategy a member of the client class?